configuretools.cmake

```diff
@@ -0,0 +1,75 @@
+# Compiler tools only have the major and minor version in their name
+if(CMAKE_CXX_COMPILER_VERSION MATCHES "^([0-9]+).([0-9]+)")
+  set(CLR_CMAKE_COMPILER_TOOL_VERSION "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}")
+else()
+  set(CLR_CMAKE_COMPILER_TOOL_VERSION ${CMAKE_CXX_COMPILER_VERSION})
+endif()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  if(APPLE)
+    set(LLVM_PREFIX "")
+  else()
+    set(LLVM_PREFIX "llvm-")
+  endif()
+  function(locate_llvm_exec exec var)
```

Comment:
Should we move this to functions.cmake?

Reply:
Since the `locate_llvm_exec` and `locate_gcc_exec` files are only used in this file, I think it might be worth keeping localized to this file. Also, most if not all of the functions in `functions.cmake` don't depend on global state  (other than some of the install-related commands).